packages/format-library/src/link/index.js

```diff
@@ -64,6 +65,18 @@ function Edit( {
 
 	function stopAddingLink( returnFocus = true ) {
 		setAddingLink( false );
+
+		// Remove any active formats.
+		// Cloning the existing value has the effect of removing
+		// any active formats and is roughly equivalent to the
+		// following mutation:
+		// value.activeFormats = [];
+		const newValue = concat( value );
+
+		// Update to force format to no longer be active.
+		onChange( newValue );
```

Comment:
This is not a good idea as it triggers changes to the value which causes the Post which the link is within to become dirty.

We will need another way to achieve this.